problems/service-placement/problem-statement-sp.md

```diff
@@ -0,0 +1,146 @@
+<!--
+SPDX-FileCopyrightText: 2024 Noé Godinho <noe@dei.uc.pt>
+
+SPDX-License-Identifier: CC-BY-4.0
+-->
+
+# Service Placement Problem
+
+Noé Godinho, University of Coimbra, Centre for Informatics and Systems of the University of Coimbra, Department of Informatics Engineering.
+
+Copyright 2024 ... Noé Godinho.
+
+This document is licensed under CC-BY-4.0.
+
+## Introduction
+
+One of the main problems in communications is the routing problem, where the goal is to find a path between two devices that minimises a given metric. Usually, that metric is latency or number of hops. However, that problem is too restrictive, due to bandwidth or device usage not being taken into account when solving the problem.
+
+The service placement problem was proposed to tackle these issues. The goal is to optimise a set of requests in a network, where each request is composed by source and destination, bandwidth requirements and a set of services to offload, with CPU and RAM requirements. This problem is NP-complete, making it hard to solve in realistic scenarios. Thus, efficient techniques to obtain (near-)optimal solutions are necessary. 
+
+## Task
+
+Given a graph, a set of requests and their services to be offloaded, minimise the overall latency and processing time of all requests and services, taking resource constraints into account.
+
+## Detailed description
+
+Given **N** nodes, **E** edges, **R** requests and **S$_r$** services for each request $r$, the following objective function aims to minimise latency and processing time:
+
+$\min \sum_{r \in R} \left(\sum_{(u,v) \in E} x_{uvr}l_{uv} + \sum_{i \in N} \sum_{s \in S_r} y_{irs}p_i \right)$
+
+where $x_{er}$ is a 1 if request $r$ traverses edge $e$, 0 otherwise. $y_{irs}$ is 1 if service $s$ from request $r$ is offloaded to node $i$, 0 otherwise. $l_e$ and $p_i$ are the latency of edge $e$ and processing time of node $i$, respectively.
+
+The first constraints aim to ensure that the offloaded services do not exceed the total CPU and RAM. The equations are as follows:
+
+$\sum_{r \in R} \sum_{s \in S_r} y_{irs} cpu_{rs} \leq C_i, ~~~~~\forall i \in N$
+$\sum_{r \in R} \sum_{s \in S_r} y_{irs} ram_{rs} \leq M_i, ~~\forall i \in N$

```

Comment:
Since $N$ is the number of nodes, and not a set, I find that the notation $i \in N$ might be confusing to a reader. One possibility would be to use $\forall i \in \\{1,\ldots,N\\}$ (`$\forall i \in \\{1,\ldots,N\\}$`).

However, a possibly better alternative is to say in the first sentence that $N$ is the set of nodes (not the number) and then use $\|N\|$ is you need the size of the set. This solution might be easier to read overall and less changes will be needed since this is happening in quite a few places.